### DISCUSSION TITLE
Strange wav sound from short text

### DISCUSSION THREAD

**Discussion by Hk1020** [coqui-ai/TTS#2918] [General Q&A]
### Describe the bug

I am just a simple user who wants to use coqui for doing text to speech. So I entered
`tts --model_name tts_models/en/ljspeech/tacotron2-DDC --out_path out.wav --text 'Another way is'`
and get a very interesting sound lasting1:56. She kind of starts singing. Different text seems to work fine.

And if this is the wrong place to ask this question please let me know where I should ask,

### To Reproduce

tts --model_name tts_models/en/ljspeech/tacotron2-DDC --out_path out.wav --text 'Another way is'

### Expected behavior

A wav file of one or two seconds length saying "Another way is"

### Logs

```shell
vega> tts --model_name tts_models/en/ljspeech/tacotron2-DDC --out_path out.wav --text 'Another way is'
 > tts_models/en/ljspeech/tacotron2-DDC is already downloaded.
 > vocoder_models/en/ljspeech/hifigan_v2 is already downloaded.
 > Using model: Tacotron2
 > Setting up Audio Processor...
 | > sample_rate:22050
 | > resample:False
 | > num_mels:80
 | > log_func:np.log
 | > min_level_db:-100
 | > frame_shift_ms:None
 | > frame_length_ms:None
 | > ref_level_db:20
 | > fft_size:1024
 | > power:1.5
 | > preemphasis:0.0
 | > griffin_lim_iters:60
 | > signal_norm:False
 | > symmetric_norm:True
 | > mel_fmin:0
 | > mel_fmax:8000.0
 | > pitch_fmin:1.0
 | > pitch_fmax:640.0
 | > spec_gain:1.0
 | > stft_pad_mode:reflect
 | > max_norm:4.0
 | > clip_norm:True
 | > do_trim_silence:True
 | > trim_db:60
 | > do_sound_norm:False
 | > do_amp_to_db_linear:True
 | > do_amp_to_db_mel:True
 | > do_rms_norm:False
 | > db_level:None
 | > stats_path:None
 | > base:2.718281828459045
 | > hop_length:256
 | > win_length:1024
 > Model's reduction rate `r` is set to: 1
 > Vocoder Model: hifigan
 > Setting up Audio Processor...
 | > sample_rate:22050
 | > resample:False
 | > num_mels:80
 | > log_func:np.log
 | > min_level_db:-100
 | > frame_shift_ms:None
 | > frame_length_ms:None
 | > ref_level_db:20
 | > fft_size:1024
 | > power:1.5
 | > preemphasis:0.0
 | > griffin_lim_iters:60
 | > signal_norm:False
 | > symmetric_norm:True
 | > mel_fmin:0
 | > mel_fmax:8000.0
 | > pitch_fmin:1.0
 | > pitch_fmax:640.0
 | > spec_gain:1.0
 | > stft_pad_mode:reflect
 | > max_norm:4.0
 | > clip_norm:True
 | > do_trim_silence:False
 | > trim_db:60
 | > do_sound_norm:False
 | > do_amp_to_db_linear:True
 | > do_amp_to_db_mel:True
 | > do_rms_norm:False
 | > db_level:None
 | > stats_path:None
 | > base:2.718281828459045
 | > hop_length:256
 | > win_length:1024
 > Generator Model: hifigan_generator
 > Discriminator Model: hifigan_discriminator
Removing weight norm...
 > Text: Another way is
 > Text splitted to sentences.
['Another way is']
   > Decoder stopped with `max_decoder_steps` 10000
 > Processing time: 72.94576692581177
 > Real-time factor: 0.6252348480556914
 > Saving output to out.wav

```
```


### Environment

shell
{
    "CUDA": {
        "GPU": [
            "NVIDIA GeForce GTX 750 Ti"
        ],
        "available": true,
        "version": "11.8"
    },
    "Packages": {
        "PyTorch_debug": false,
        "PyTorch_version": "2.0.1+cu118",
        "TTS": "0.16.5",
        "numpy": "1.22.0"
    },
    "System": {
        "OS": "Windows",
        "architecture": [
            "64bit",
            "WindowsPE"
        ],
        "processor": "Intel64 Family 6 Model 60 Stepping 3, GenuineIntel",
        "python": "3.9.13",
        "version": "10.0.19045"
    }
}
```

**Comment by AmurgCodru** [comment]
I generated a long audio (30 minutes) and I got stuttering multiple times which is awkward, maybe related to this issue

**Comment by nsqning** [comment]
I met this issue as well, it generates a stutter wav([here](https://github.com/nsqning/open_issues/blob/main/tts/stutter.wav))

**Comment by Hk1020** [comment]
This looks very similar. Your result is also 1:59 minutes long - just as mine. The actual text is vocalized completely but kind of bad and then there starts a lot of nonsense. Could it be some kind of hallucination?

**Comment by erogol** [comment]
it may happen. Requires training with longer sentences. Unfortunately, there is not easy fix. You should try a different model.

**Reply by Hk1020** [reply]
Interesting, thanks for the reply. It's rather the short "sentences" like my three word example which seem to trigger it. 

I am using tts from a Python script and I am wondering how I can detect when this has happened. I am completely new to Python and all I see is the error message with max_steps on stdout. Is there some return value from tts-to_file I could act on?

**Reply by FurkanGozukara** [reply]
do you suggest longer for fine tuning? like 15 seconds?

**Reply by erogol** [reply]
train with whatever it doesn't work for. I am not sure 15 is the value but something about it.

**Reply by FurkanGozukara** [reply]
the thing is i am not sure whatever won't work. I am trying to prepare a tutorial for general public - usually not programmers

**Reply by xingyuhe23** [reply]
i have had the same issue but i do resolve this issue by add a dot at the end of the sentence see if this works for you.   I believe that you will need to provide the model a clear ending so that  it  know where to end and not keep generating strange noise
